src/argilla/client/feedback/schemas/remote/metadata.py

```diff
@@ -28,13 +28,48 @@
 if TYPE_CHECKING:
     import httpx
 
-    from argilla.client.feedback.schemas.types import AllowedMetadataPropertyTypes
+    from argilla.client.feedback.schemas.types import AllowedMetadataPropertyTypes, AllowedRemoteMetadataPropertyTypes
     from argilla.client.sdk.v1.datasets.models import FeedbackMetadataPropertyModel
 
 
 class _RemoteMetadataProperty(RemoteSchema):
     @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
-    def delete(self) -> "AllowedMetadataPropertyTypes":
+    def update(
+        self: "AllowedRemoteMetadataPropertyTypes",
+        title: Optional[str] = None,
+        visible_for_annotators: Optional[bool] = None,
+    ) -> None:
+        """Updates the `RemoteMetadataProperty` in Argilla.
+
+        Args:
+            title: the new title of the metadata property. Defaults to `None`.
+            visible_for_annotators: whether the metadata property should be visible for annotators. Defaults to `None`.
+
+        Returns:
+            The updated `RemoteMetadataProperty` object.
+        """
+        if title is None and visible_for_annotators is None:
+            raise ValueError("At least one of `title` or `visible_for_annotators` must be provided and not `None`.")
+        if title is not None and title == self.title:
+            raise ValueError(
+                f"Provided `title={title}` is the same as the current one `{self.title}`, you need either to provide"
+                " a different one, or just pass it empty."
+            )
+        if visible_for_annotators is not None and visible_for_annotators == self.visible_for_annotators:
+            raise ValueError(
+                f"Provided `visible_for_annotators={visible_for_annotators}` is the same as the current one"
+                f" `{self.visible_for_annotators}`, you need either to provide a different one, or just pass it empty."
+            )
```

Comment:
I think we should just skip the remote update when the original and new data are the same. We should avoid side effects as much as possible. For example, running a property update twice, the first will do an update but the second will raise an error, which can be confusing to the user.

Just skipping the remote update when data remains the same is a good optimization, but not a reason to raise an error IMHO